### PR TITLE
feat: Replaced Image.network with CachedNetworkImage

### LINKED ISSUE #255: feat: Replace `Image.network` with CachedNetworkImage.
**Description**

This issue is set to open discussion for replacing `Image.network` with [cached_network_image](https://pub.dev/packages/cached_network_image).

This will add a dependency on a third-party package but it will help a lot in reducing the network load.

### PR DESCRIPTION
**Pull request by yousufmyk**
## Description

Replaced Image.network with CachedNetworkImage 

## Related Issue 

closes #255 

## Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code refactor
- [ ] Build configuration change
- [ ] Documentation
- [ ] Chore

### CODE REVIEW

File: packages/mirai/lib/src/parsers/mirai_image/mirai_image_parser.dart [hunk 1]
```diff
@@ -30,17 +31,42 @@ class MiraiImageParser extends MiraiParser<MiraiImage> {
...
+  //       model.src,
+  //       alignment: model.alignment.value,
+  //       color: model.color?.toColor(context),
+  //       width: model.width,
+  //       height: model.height,
+  //       fit: model.fit,
+  //       errorBuilder: (context, error, stackTrace) {
+  //         return const SizedBox();
+  //       },
+  //     );
```
Comment (divyanshub024) [inline packages/mirai/lib/src/parsers/mirai_image/mirai_image_parser.dart:56]: Can we remove these unwanted comments?

**Review by divyanshub024 — CHANGES_REQUESTED** [review]
Hey @yousufmyk 👋🏻 

Thank you for the amazing contribution. I have some suggestions, please check my comments.

**Comment by divyanshub024** [comment]
@yousufmyk There are some formatting issues in `mirai_image_parser` can you please check?

**Comment by yousufmyk** [comment]
Done, You can check now

**Comment by yousufmyk** [comment]
I see, I've tried fixing Dart formatting Issues. Hopefully, now it should work fine.

**Comment by yousufmyk** [comment]
@divyanshub024  don’t know why this keeps failing I tried "dart format --set-exit-if-changed ." and it works but again getting an issue do you have any idea about this issue?

**Comment by divyanshub024** [comment]
@yousufmyk 
Can you try running the following commands?

Make sure you have Melos installed in your system:

```
dart pub global activate melos
```
And then run
```
1. melos clean
2. melos bs
3. melos format
```

**Comment by yousufmyk** [comment]
@divyanshub024 Thank you for the suggestions!  appreciate you taking the time to help troubleshoot.

I've now installed Melos and run the commands you provide.

**Comment by yousufmyk** [comment]
I can see it’s again failing but on my side it says "
melos run format
  └> melos exec -- "dart format ."
     └> SUCCESS" after running melos

**Comment by divyanshub024** [comment]
@yousufmyk Can you run `flutter pub get` in mirai_gallery? When I ran it `pubspec.lock` file showed some changes. Alternatively, you can also run `melos bs` from the root folder.

**Comment by divyanshub024** [comment]
> @yousufmyk Can you run `flutter pub get` in mirai_gallery? When I ran it `pubspec.lock` file showed some changes. Alternatively, you can also run `melos bs` from the root folder.

Hey @yousufmyk, Did it help?

**Comment by divyanshub024** [comment]
Closed by #289

**Pull request closed without merging**